core/tee/tee_svc_storage.c

```diff
@@ -444,7 +445,8 @@ TEE_Result syscall_storage_obj_del(unsigned long obj)
 	if (o->pobj == NULL || o->pobj->obj_id == NULL)
 		return TEE_ERROR_BAD_STATE;
 
-	if (IS_ENABLED(CFG_NXP_SE05X)) {
+	/* Scan buffers that might contain a Secure Element key identifier */
+	if (IS_ENABLED(CFG_NXP_SE05X) && o->info.dataSize < SMALL_PAGE_SIZE) {
```

Comment:
I'd rather have a special define for this, with a check where these objects are written also to not exceed this.

Reply:
This is really SE05X specific.
A comment a bit out of scope of this P-R: as `crypto_storage_obj_del()` is implemented only for SE05X, could this function be renamed to a se050 API function and do what se050 driver expects, leaving this generic `syscall_storage_obj_del()` function with:
```c
	if (IS_ENABLED(CFG_NXP_SE05X))
		se050_maybe_delete_keyid(o);
```
`se050_maybe_delete_keyid()` in _core/drivers/crypto/se050/core/storage.c_ would do whatever SE050 needs to, including what `crypto_storage_obj_del()` does.

Reply:
I think `crypto_storage_obj_del` should be left generic (it ha a weak implementation).